qiskit/opflow/state_fns/state_fn.py

```diff
@@ -144,14 +144,17 @@ def _expand_dim(self, num_qubits: int) -> 'StateFn':
         raise NotImplementedError
 
     def permute(self, permutation: List[int]) -> OperatorBase:
-        """Permute the qubits of the state function.
+        """Permute the qubits of the state abstract function.
```

Comment:
I think some `StateFn`s inherit the docstring for this method (e.g. `DictStateFn` has no docstring for `permute`) and it would be nice if that didn't say "abstract". We could either add a docstring to `DictStateFn` or not put `abstact` here, I'd be good with both

Reply:
I added this because the way doc subclassing is done in the code imported from aqua is weird and results in duplicate paths that sphinx can't resolve. I was trying to fix the sphinx warnings before I had to break up the rst documents in `docs/apidocs` so I'll just revert this for now because it's not necessary to get things building (which is the point of this PR). We can look at cleaning up the docs later because I do think we should make it clear between the subclass methods and how they behave and the abstract interfaces expectations.